examples/computer_vision/contrastive_learning_pytorch/train_eval_model_def.py

```diff
@@ -0,0 +1,314 @@
+            [
+                transforms.RandomResizedCrop(
+                    size=self.data_config.crop_size, scale=(0.2, 1.0)
+                ),
+                transforms.RandomHorizontalFlip(),
+                transforms.RandomApply(
+                    [transforms.ColorJitter(0.4, 0.4, 0.4, 0.1)], p=0.8
+                ),
+                transforms.RandomGrayscale(p=0.2),
+                transforms.ToTensor(),
+                normalize,
+            ]
+        )
+        if self.data_config.dataset == "cifar10":
+            train_dataset = datasets.CIFAR10(
+                root=self.download_dir,
+                transform=utils.TwoCropTransform(train_transform),
+                download=True,
+            )
+        else:  # dataset is cifar100
+            train_dataset = datasets.CIFAR100(
+                root=self.download_dir,
+                transform=utils.TwoCropTransform(train_transform),
+                download=True,
+            )
+
+        # We will modify the dataset so that the automatically sharded datasets have the same len.
+        rounded_length = (
+            len(train_dataset) // self.context.get_global_batch_size()
+        ) * self.context.get_global_batch_size()
+        train_dataset = torch.utils.data.Subset(train_dataset, range(rounded_length))
+
+        train_loader = det_torch.DataLoader(
+            train_dataset,
+            batch_size=self.context.get_per_slot_batch_size(),
+            pin_memory=True,
+            num_workers=self.data_config.num_workers,
+            shuffle=True,
+            drop_last=True,
+        )
+        return train_loader
+
+    def build_validation_data_loader(self) -> Any:
+        assert (
+            self.data_config.dataset in STATS
+        ), "only supports cifar10 and cifar100 datasets"
+        mean = STATS[self.data_config.dataset]["mean"]
+        std = STATS[self.data_config.dataset]["std"]
+        normalize = transforms.Normalize(mean=mean, std=std)
+
+        val_transform = transforms.Compose(
+            [
+                transforms.ToTensor(),
+                normalize,
+            ]
+        )
+        if self.data_config.dataset == "cifar10":
+            val_dataset = datasets.CIFAR10(
+                root=self.download_dir,
+                train=False,
+                transform=val_transform,
+                download=True,
+            )
+        else:  # dataset is cifar100
+            val_dataset = datasets.CIFAR100(
+                root=self.download_dir,
+                train=False,
+                transform=val_transform,
+                download=True,
+            )
+
+        val_loader = det_torch.DataLoader(
+            val_dataset,
+            batch_size=self.context.get_per_slot_batch_size(),
+            pin_memory=True,
+            num_workers=self.data_config.num_workers,
+        )
+        return val_loader
+
+    def build_train_cls_data_loader(self) -> Any:
+        assert (
```

Comment:
This seems to be the exact same code as build_train_data_loader. Lets reuse code instead of having it duplicated.